amrendra-rathore/express-server
# Day5

### Task Description:

Install following packages as dev dependencies:
- @babel/core
- @babel/node
- @babel/preset-env

Create .babelrc file with the following content

~~~
{
"presets": ["@babel/preset-env"]
}
~~~


Add a file .gitignore in the root directory and add node_modules and package-lock.json into it.

Create a file index.js inside the extra folder.
Running the file index.js with babel and also write a script for the same in package.json.


The command will be:
~~~
npx babel-node extra
~~~


**Patterns:**
Export the functions of the patterns from each file(using export default), Also removing function calling from these files.
Import all the patterns inside patterns/index.js and again export all the patterns from the index.js file.
Import all the functions in extra/index.js exported from extra/patterns/index.js and now execute these functions with various arguments.

**Utils:**
-Permission:
* Create a file 'constants' inside folder extra.
* Move all the constant values to the above file.
* Export all constants using export.
* Remove hasPermission callings from permission.js.
* Export function hasPermission using export default.
* Import the hasPermission inside utils/index.js and export again using export.
* Import hasPermission inside extra/index.js and call the function with various arguments.

-Validation:
* Create a file helpers.js inside the utils folder.
* Move the function validateEmail from validation.js to helpers.js and export it using export.
* Import validateEmail from helpers.js inside validation.js.
* Export and import the function the same as done for hasPermission above.
* Move the constant users and function calling from validation.js to extra/index.js.


Proper logging should be there before each and every function being called in extra/index.js
